app/components/search_input_component/search_input_component.html.haml

```diff
@@ -0,0 +1,5 @@
+= component_controller do
+  %div.search-input
+    %input{type: 'text', placeholder: t("components.search_input.placeholder"), id: 'search_query', data: {action: 'debounced:input->search-input#search'}, value: @value}
+    .search-button{data: @data}
```

Comment:
I'm used to hit `enter` to submit a form but it didn't work here. Does it need a submit button?

Reply:
Done!

```js
  connect() {
    super.connect();
    this.element
      .querySelector("input")
      .addEventListener("keydown", this.searchOnEnter);
  }

  disconnect() {
    super.disconnect();
    this.element
      .querySelector("input")
      .removeEventListener("keydown", this.searchOnEnter);
  }

  searchOnEnter = (e) => {
    if (e.key === "Enter") {
      this.element.querySelector(".search-button").click();
    }
  };
```